ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/LoggingRateLimiter.java

```diff
@@ -45,13 +56,33 @@
     requireNonNull(ksqlRestConfig);
     this.rateLimiterFactory = requireNonNull(rateLimiterFactory);
     this.rateLimitedPaths = getRateLimitedRequestPaths(ksqlRestConfig);
+    this.rateLimitedResponseCodes = getRateLimitedResponseCodes(ksqlRestConfig);
+    this.pathLimitHit = rateLimiterFactory.apply(LIMIT_HIT_LOG_RATE);
+    this.responseCodeLimitHit = rateLimiterFactory.apply(LIMIT_HIT_LOG_RATE);
   }
 
-  public boolean shouldLog(final String path) {
+  public boolean shouldLog(final Logger logger, final String path, final int responseCode) {
     if (rateLimitedPaths.containsKey(path)) {
       final double rateLimit = rateLimitedPaths.get(path);
-      rateLimiters.computeIfAbsent(path, (k) -> rateLimiterFactory.apply(rateLimit));
-      return rateLimiters.get(path).tryAcquire();
+      rateLimitersByPath.computeIfAbsent(path, (k) -> rateLimiterFactory.apply(rateLimit));
```

Comment:
Would it be better to initialize the rate limiters up front, rather than calling `computeIfAbsent` on each request? I don't have a sense of how large this optimization is, feels minor but I also feel it can't hurt. Feel free to disagree.

Reply:
Yeah, you're right.  The data is static and not large, so why not just make an immutable map instead on initialization?

I was originally thinking it might be dynamic, but it's not currently.